src/ucp/core/ucp_ep.inl

```diff
@@ -25,19 +25,28 @@ static inline ucp_ep_config_t *ucp_ep_config(ucp_ep_h ep)
 }
 
 static UCS_F_ALWAYS_INLINE uct_ep_h ucp_ep_get_lane(ucp_ep_h ep,
-                                                    ucp_lane_index_t lane_index)
+                                                    ucp_lane_index_t lane_index,
+                                                    int fast_path)
 {
     ucs_assert(lane_index != UCP_NULL_LANE);
 
-    return ep->uct_eps[lane_index];
+	if (fast_path || (lane_index < UCP_MAX_FAST_PATH_LANES)) {
+		return ep->uct_eps[lane_index];
```

Comment:
1. if fast-path set to 1, assert that lane < UCP_MAX_FAST_PATH_LANES
2. otherwise, assert that lane < UCP_MAX_LANES